File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

bool isValid(int n);
void repeatedChars(int n, char c);


// This code is for the harder mario problem: https://cs50.harvard.edu/x/2020/psets/1/mario/more/.
int main(void)
{
    bool valid;
    int height;

    // Keep prompting for a number that is between 1 and 8 inclusive.
    do
    {
        height = get_int("Height: ");
        valid = isValid(height);

    }
    while (valid == 0);

    // Find the length of the base string (including spaces).
    int base_length = 2 + 2 * height;

    // Counter for each level
    for (int current_height = 1; current_height <= height; current_height++)
    {
        int num_brick_left = current_height;
        int num_space_left = height - current_height;

        repeatedChars(num_space_left, ' ');
        repeatedChars(num_brick_left, '#');
        repeatedChars(2, ' ');
        repeatedChars(num_brick_left, '#');
        printf("\n");
    }
}

// Prints the string with the character argument repeated int n times.
void repeatedChars(int n, char c)
{
    for (int i = 0; i < n; i++)
    {
        printf("%c", c);
    }
    return;
}


// Returns true if the number is valid ie between 1 and 8 inclusive.
bool isValid(int n)
{
    if ((n <= 8) && (n >= 1))
    {
        return true;
    }
    else
    {
        return false;
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    # Get valid input from user
    height = -1
    while (True):
        height = get_int("Height: ")

        if (is_valid(height) == True):
            break

    # print("Valid n is " + str(n))

    # Start printing the pyramids

    # Find the length of the base string (including spaces).
    base_length = 2 + 2 * height

    # Counter for each level
    for current_height in range(1, height + 1):
        num_brick_left = current_height
        num_space_left = height - current_height

        print_repeated_char(num_space_left, ' ')
        print_repeated_char(num_brick_left, '#')
        print_repeated_char(2, ' ')
        print_repeated_char(num_brick_left, '#')
        print("")


'''
Checks if the argument int is valid
Returns True if between 1 and 8 inclusive
Returns False if not between 1 and 8 inclusive
'''


def is_valid(num):
    if (num < 1) or (num > 8):
        return False
    else:
        return True

# Helper function to print a single char 'n' number of times


def print_repeated_char(n, char):
    for i in range(n):
        print(char, end="")


if __name__ == "__main__":
    main()